vita3k/gui/src/content_manager.cpp

```diff
@@ -83,94 +85,119 @@ struct SaveData {
 };
 
 static std::vector<SaveData> save_data_list;
+static std::map<std::string, std::string> space;
+static std::map<std::string, bool> contents_selected, finished_get_size;
+static size_t apps_list_size, save_data_list_size;
 
 static void get_save_data_list(GuiState &gui, HostState &host) {
     host.io.user_id = fmt::format("{:0>2d}", host.cfg.user_id);
     save_data_list.clear();
+    save_data_list_size = 0;
+
+    std::thread get_save_data_size([&gui, &host]() {
```

Comment:
Full of race conditions. You wouldn't want to pull in the host here.

Reply:
No understund, i need host here

Reply:
I mean some functions are not thread safe. You should consider it.

Reply:
Yes i have see that forbsome thinks

Reply:
i have added this mutex if you fcan check again please